packages/headless/src/database/mongodb/wishList/index.ts

```diff
@@ -0,0 +1,123 @@
+import { Injectable } from '@nestjs/common';
+import { Item, WishList } from 'src/entity/wishList';
+import { IWishListDatabase } from 'src/modules/wishlist/repositories/wishList.database.interface';
+import { WishListModel } from './wishList.model';
+@Injectable()
+export class WishListDatabase implements IWishListDatabase {
+  async findWishListByUserId(userId: string): Promise<WishList | null> {
+    const wishList = await WishListModel.findOne({ user: userId }).lean();
+    return Promise.resolve(wishList);
+  }
+
+  async findItemByUserIdAndProductId(userId: string, productId: string,): Promise<WishList | null> {
+    const wishList = await WishListModel.findOne({
+      user: userId,
+      'items.product': productId,
+    }).lean();
+    return Promise.resolve(wishList);
+  }
+
+  async findWishListAndIncrementItem(userId: string, item: Item,): Promise<WishList | null> {
```

Comment:
You can rename the function as ```incrementWishListItem``` because you are gonna increment item to wishlist